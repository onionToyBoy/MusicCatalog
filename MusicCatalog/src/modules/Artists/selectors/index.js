export const selectArtists = state => state.search.artists.results;

export const selectAlbums = state => state.goTo.albums.results;

export const selectTracks = state => state.goTo.tracks.results;
