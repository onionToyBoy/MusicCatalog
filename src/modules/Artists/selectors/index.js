export const selectArtists = state => state.artistsReducer.artists;

export const selectAlbums = artistId => state => state.artistsReducer.albums[artistId];

export const selectTracks = albumId => state => state.artistsReducer.tracks[albumId];
