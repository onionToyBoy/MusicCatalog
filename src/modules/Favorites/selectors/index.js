export const selectFavorites = state => state.favoriteAlbumsReducer.favorites;
