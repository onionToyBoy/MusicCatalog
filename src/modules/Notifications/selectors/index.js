export const selectIsLoading = state => state.notificationReducer.isLoading;

export const selectIsError = state => state.notificationReducer.isError;
