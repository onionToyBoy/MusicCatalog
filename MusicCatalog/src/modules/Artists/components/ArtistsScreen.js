import React, { useEffect, useState } from 'react';
import { ActivityIndicator, FlatList, View, StyleSheet } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { Navigation } from 'react-native-navigation';

import { colors } from '../../../constants/colors';
import { Artist } from './Artist';
import { selectArtists } from '../selectors';
import { searchArtist } from '../thunks';
import { routes } from '../../../constants/routes';
import { SearchBar } from '../../../components/SearchBar';

export const ArtistsScreen = ({ componentId }) => {
  const [searchValue, setSearchValue] = useState('artists');

  const dispatch = useDispatch();

  const artists = useSelector(selectArtists);

  useEffect(() => {
    dispatch(searchArtist(searchValue));
  }, [searchValue,dispatch]);

  const toAlbums = ( name, id) => {
    Navigation.push(componentId, {
      component: {
        name: routes.ArtistsAlbums,
        id: id,
        options: {
          topBar: {
            title: {
              text: name+' albums',
            },
          },
        },
      },
    });
  };

  const renderArtists = ( {item} ) => (
    <Artist
      name={item.artistName}
      genre={item.primaryGenreName}
      openAlbums={toAlbums}
      id={item.artistId}
    />
  )

  return (
    <View style={styles.container}>
      <SearchBar onSearch={setSearchValue} />
      <FlatList
        data={artists}
        keyExtractor={item => item.artistId}
        renderItem={renderArtists}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 5,
    backgroundColor: colors.DARK_GRAY,
  },
});
