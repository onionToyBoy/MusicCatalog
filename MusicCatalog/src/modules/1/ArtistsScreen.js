import React, {useEffect, useState} from 'react';
import {
  ActivityIndicator,
  FlatList,
  Text,
  View,
  StyleSheet,
} from 'react-native';
import { SearchBar } from '../../components/SearchBar';

import {colors} from '../../constants/colors';

export const ArtistsScreen = () => {
  const [isLoading, setLoading] = useState(true);
  const [data, setData] = useState([]);

  let tmpArray = [];

  const itemCheck = item => {
    if (tmpArray.indexOf(item.artistName) === -1) {
      tmpArray.push(item.artistName);
      return true;
    }
    return false;
  };

  useEffect(() => {
    fetch(
      'https://itunes.apple.com/search?term=artists&media=music&entity=musicArtist&limit=20',
    )
      .then(response => response.json())
      .then(json => setData(json.results.filter(item => itemCheck(item))))
      .catch(error => console.error(error))
      .finally(() => setLoading(false));
  }, []);

  return (
    <View style={styles.container}>
    <SearchBar/>
      {isLoading ? (
        <ActivityIndicator />
      ) : (
        <FlatList
          data={data}
          keyExtractor={item => item.artistId}
          renderItem={({item}) => (
            <Text style={styles.text}>
              {item.artistName}, {item.primaryGenreName}
            </Text>
          )}
        />
      )}
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 5,
    backgroundColor: colors.DARK_GRAY,
  },
  text: {
    color: colors.YELLOW,
    fontSize: 15,
    marginVertical: 10,
  },
});
