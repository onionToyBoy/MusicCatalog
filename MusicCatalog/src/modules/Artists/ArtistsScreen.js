import React, { useEffect, useState } from 'react';
import { ActivityIndicator, FlatList, View, StyleSheet } from 'react-native';
import { SearchBar } from '../../components/SearchBar';
import { useDispatch, useSelector } from 'react-redux';

import { colors } from '../../constants/colors';
import { Artist } from './components/Artist';
import { selectArtists } from './selectors';
import { searchArtist } from './thunks';
import { searchChanged } from './actions/searchChanged';

export const ArtistsScreen = () => {
  //const [isLoading, setLoading] = useState(true);
  const [searchValue, setSearchValue] = useState('artists');

  const dispatch = useDispatch();

  const artists = useSelector(selectArtists);

  // const tmpArray = [];

  // const itemCheck = item => {
  //   if (tmpArray.indexOf(item.artistName) === -1) {
  //     tmpArray.push(item.artistName);
  //     return true;
  //   }
  //   return false;
  // };

  useEffect(() => {
    dispatch(searchArtist(searchValue));
  }, [searchValue]);

  const openAlbums = name => {
    alert(name);
  };

  const searching = text => {
    setSearchValue(text);
  };

  return (
    <View style={styles.container}>
      <SearchBar onSearch={searching} />

      <FlatList
        data={artists}
        keyExtractor={item => item.artistId}
        renderItem={({ item }) => (
          <Artist name={item.artistName} genre={item.primaryGenreName} openAlbums={openAlbums} />
        )}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 5,
    backgroundColor: colors.DARK_GRAY,
  },
});
