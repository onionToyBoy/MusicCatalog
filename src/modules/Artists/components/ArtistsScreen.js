import React, { useEffect, useState } from 'react';
import { FlatList, View, StyleSheet } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { Navigation } from 'react-native-navigation';

import { colors } from '../../../constants/colors';
import { Artist } from './Artist';
import { selectArtists } from '../selectors';
import { searchArtist } from '../thunks';
import { routes } from '../../../constants/routes';
import { SearchBar } from '../../../components/SearchBar';

export const ArtistsScreen = ({ componentId }) => {
  const [searchValue, setSearchValue] = useState('artists');

  const dispatch = useDispatch();

  const artists = useSelector(selectArtists);

  useEffect(() => {
    dispatch(searchArtist(searchValue));
  }, [searchValue, dispatch]);

  const onOpenAlbum = (name, id) => {
    Navigation.push(componentId, {
      component: {
        name: routes.ArtistsAlbums,
        id: id,
        passProps: {
          artistId: id,
        },
        options: {
          topBar: {
            title: {
              text: `${name} albums`,
            },
          },
        },
      },
    });
  };

  const renderArtists = ({ item }) => <Artist {...item} onOpenAlbum={onOpenAlbum} />;

  return (
    <View style={styles.container}>
      <SearchBar onSearch={setSearchValue} />
      <FlatList data={artists} keyExtractor={item => item.artistId} renderItem={renderArtists} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 5,
    backgroundColor: colors.DARK_GRAY,
  },
});
