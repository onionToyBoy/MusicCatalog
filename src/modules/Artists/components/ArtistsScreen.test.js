import React from 'react';
import { shallow } from 'enzyme';

import { ArtistsScreen } from './ArtistsScreen';
import * as NetinfoModule from '@react-native-community/netinfo';
import * as SelectorsModule from '../selectors';
import { Navigation } from 'react-native-navigation';

jest.mock('@react-native-community/netinfo', () => ({
  useNetInfo: jest.fn(() => ({ isConnected: true })),
}));
jest.mock('react-native-navigation', () => ({
  Navigation: {
    push: jest.fn((componentId, obj) => {}),
  },
}));
jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
  useSelector: selector => selector(),
}));

describe('ArtistsScreen test', () => {
  const componentId = '123';
  const artists = [
    { artistName: 'Rob Zombie', artistId: 110374 },
    { artistName: 'Cypres hill', artistId: 1449305168 },
    { artistName: 'Navel', artistId: 1530211376 },
  ];
  const navigationObject = {
    component: {
      name: 'ArtistsAlbums',
      options: { topBar: { title: { text: 'undefined albums' } } },
      passProps: { artistId: undefined },
    },
  };
  beforeEach(() => {
    jest.spyOn(SelectorsModule, 'selectArtists').mockImplementation(() => [artists]);
    jest.spyOn(NetinfoModule, 'useNetInfo').mockImplementation(() => ({ isConnected: true }));
  });

  test(' Renders warning notification when isConnected is false', () => {
    jest.spyOn(NetinfoModule, 'useNetInfo').mockImplementation(() => ({ isConnected: false }));
    const wrapper = shallow(<ArtistsScreen componentId={componentId} />);

    expect(wrapper.find({ testId: 'Warning-notification' })).toHaveLength(1);
    expect(wrapper.find({ testId: 'Welcome-notification' })).not.toHaveLength(1);
  });

  test(' Renders welcome notification when isConnected is true', () => {
    const wrapper = shallow(<ArtistsScreen componentId={componentId} />);

    expect(wrapper.find({ testId: 'Warning-notification' })).not.toHaveLength(1);
    expect(wrapper.find({ testId: 'Welcome-notification' })).toHaveLength(1);
  });

  test('New test', () => {
    const wrapper = shallow(<ArtistsScreen componentId={componentId} />);
    const newValue = 'str';

    wrapper.find('SearchBar').prop('onSearch')?.(newValue);

    expect(wrapper.find('SearchBar').prop('searchValue')).toBe(newValue);
  });

  test('On clearInput', () => {
    const wrapper = shallow(<ArtistsScreen componentId={componentId} />);
    const newValue = 'str';

    wrapper.find('SearchBar').prop('onSearch')?.(newValue);
    expect(wrapper.find('SearchBar').prop('searchValue')).toBe(newValue);

    wrapper.find('SearchBar').prop('clearInput')?.();
    expect(wrapper.find('SearchBar').prop('searchValue')).toBe('');
  });

  test('FlatList', () => {
    jest.spyOn(SelectorsModule, 'selectArtists').mockImplementation(() => artists);

    const wrapper = shallow(<ArtistsScreen componentId={componentId} />);
    const newValue = 'str';
    wrapper.props().children[0].props.onSearch('sdsdsd');
    wrapper.find('SearchBar').prop('onSearch')?.(newValue);
    const flatList = wrapper.find('FlatList');
    const artist = flatList.props().renderItem(flatList.props().data[0]);
    artist.props.onOpenAlbum();

    expect(Navigation.push).toHaveBeenCalledWith(componentId, navigationObject);
  });
});
