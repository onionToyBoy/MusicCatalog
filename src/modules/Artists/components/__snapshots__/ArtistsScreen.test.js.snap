// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsScreen test FlatList 1`] = `ShallowWrapper {}`;

exports[`ArtistsScreen test On clearInput 1`] = `
<view
  style="flex:1;padding:5px;background-color:#1E1E1E"
>
  <view
    style="flex-direction:row;min-height:40px;margin-vertical:10px;border-width:1px;border-color:#505050"
  >
    <textinput
      maxlength="30"
      placeholder="Enter text..."
      placeholdertextcolor="#505050"
      style="flex:1;color:#FFFFFF;padding-horizontal:15px;font-size:17px"
      underlinecolorandroid="transparent"
      value=""
    />
    <view
      focusable="true"
      style="padding-horizontal:12px;opacity:1"
    >
      <text
        style="color:#1E1E1E"
      >
        ᳵ
      </text>
    </view>
  </view>
  <view
    style="flex:1;justify-content:center;align-items:center"
  >
    <text
      style="0:[object Object];1:[object Object]"
    >
      ♫
    </text>
    <text
      style="0:[object Object];1:[object Object]"
    >
      Search your artist
    </text>
  </view>
</view>
`;
